From 943b4e1673dcc2c4d0a2923b337338e26b022c63 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm@cyngn.com>
Date: Mon, 27 Jun 2016 14:31:57 -0700
Subject: [PATCH 06/11] Settings: Set root access options appropriately

It is possible to be running a user build with a debuggable boot image.
In this case, "su" will not be available.  So only show none/adb.

Issue-Id: BACON-4461
Change-Id: Iaa7df8311b9ea81eabb1566ba6f9159fdc9fab34
---
 res/values/cm_arrays.xml                                      | 10 ++++++++++
 res/xml/development_prefs.xml                                 |  4 +---
 src/com/android/settings/development/DevelopmentSettings.java |  9 +++++++++
 3 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/res/values/cm_arrays.xml b/res/values/cm_arrays.xml
index 5f64ea3..011ba45 100644
--- a/res/values/cm_arrays.xml
+++ b/res/values/cm_arrays.xml
@@ -217,4 +217,14 @@
         <item>2</item>
         <item>3</item>
     </string-array>
+
+    <string-array name="root_access_entries_adb" translatable="false">
+        <item>@string/root_access_none</item>
+        <item>@string/root_access_adb</item>
+    </string-array>
+
+    <string-array name="root_access_values_adb" translatable="false">
+        <item>0</item>
+        <item>2</item>
+    </string-array>
 </resources>
diff --git a/res/xml/development_prefs.xml b/res/xml/development_prefs.xml
index a03f263..032f94f 100644
--- a/res/xml/development_prefs.xml
+++ b/res/xml/development_prefs.xml
@@ -124,9 +124,7 @@
     <ListPreference
         android:key="root_access"
         android:title="@string/root_access"
-        android:persistent="false"
-        android:entries="@array/root_access_entries"
-        android:entryValues="@array/root_access_values" />
+        android:persistent="false" />
 
     <Preference
         android:key="root_appops"
diff --git a/src/com/android/settings/development/DevelopmentSettings.java b/src/com/android/settings/development/DevelopmentSettings.java
index 520d1e8..d7b83fa 100644
--- a/src/com/android/settings/development/DevelopmentSettings.java
+++ b/src/com/android/settings/development/DevelopmentSettings.java
@@ -108,6 +108,8 @@ import com.android.settingslib.drawer.CategoryKey;
 
 import lineageos.providers.LineageSettings;
 
+import org.lineageos.internal.util.FileUtils;
+
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashSet;
@@ -631,6 +633,13 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
         mRootAppops.setOnPreferenceClickListener(this);
 
         if (!removeRootOptionsIfRequired()) {
+            if (FileUtils.fileExists("/system/xbin/su")) {
+                mRootAccess.setEntries(R.array.root_access_entries);
+                mRootAccess.setEntryValues(R.array.root_access_values);
+            } else {
+                mRootAccess.setEntries(R.array.root_access_entries_adb);
+                mRootAccess.setEntryValues(R.array.root_access_values_adb);
+            }
             mAllPrefs.add(mRootAccess);
             mAllPrefs.add(mRootAppops);
         }
-- 
2.7.4

